qiskit/backends/basejob.py

```diff
@@ -0,0 +1,58 @@
+from abc import ABC, abstractmethod
+
+class BaseJob(ABC):
+    """Class to handle asynchronous jobs"""
+    
+    @abstractmethod
+    def __init__(self):
+        """Initializes and initates the asynchronous job"""
+        pass
+
```

Comment:
can we also have methods to get back `job_id` and `backend` that runs the job?

Reply:
So what's the use-case for exposing the ID? One of the points of having an abstraction over jobs is that users shouldn't need to know anything about implementation details as the Job ID, which is exclusively for remote (not local) queue based backends.

Reply:
In case we really need to expose the ID, maybe a better place could be our own implementation (IBMQJob) and not pollute the BaseJob class. If we find out that third-party implementers also need to expose the ID, then yeah, let's move it to the BaseJob class.

Reply:
I also think it's important that we try to keep the `BaseJob` as generic as possible - during our meeting, my mention of including `id` as a reachable property was aimed at the `IBMQ*` level, and provided that we had a clear current use case were users needed access to it (which I thought we did - but actually, it would be great to have a solid example of it).

Can we keep `BaseJob` id-agnostic as possible, following the idea that it should be as less restrictive and away from implementation details as possible?

Reply:
I think this sounds fine to me. Currently the spec says the ID is reported in the `status` dictionary. I suppose IBMQJob could add an additional method to have `id` or `get_id` but am somewhat concerned about adding methods which because of their common usage, are assumed to exist for all backends.